i-VRESSE/haddock3-webapp
# 63 GitHub auth

# Enable github login method

**Note! If the specific email is used to create local login, it is possible to use github login registered with the same email. But if the specific email is registered for the first time using github login, the local login with the same email is not possible.**

Changes made:
- added env variables to example .env file
- added github dependency `remix-auth-github`
- upgraded drizzle to 0.30 to support "upsert" method
- enforces rounded image in the nav
- fixed markup error ul cannot be child of p element
- created login_failed page

How to test
- Define github app and obtain client id and client secret. 
- Add this info to .env file 
- Start app, Github login method should appear (see image 1)
- After clicking to Github login you will need to accept terms (only first time)
- When using github login next time you will be logged n

## Example github login enabled
![image](https://github.com/i-VRESSE/haddock3-webapp/assets/9204081/52d23534-729f-494a-a1e1-bbb5b567c04e)

## Logged in example (avatar top right)
![image](https://github.com/i-VRESSE/haddock3-webapp/assets/9204081/f0e867f4-b872-4b50-aa4e-373650281e9b)

## Login failed page
![image](https://github.com/i-VRESSE/haddock3-webapp/assets/9204081/71c32a3c-ff18-450e-adc8-1712ec0e91f8)
